From fb455cef8a72ee97e9aba0c42d82782196c43218 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20=C4=8Cerm=C3=A1k?= <dcermak@suse.com>
Date: Mon, 20 Nov 2023 09:28:06 +0100
Subject: [PATCH] set makefile target entrypoint.gz as .PHONY on non x86_64
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The target internal/mkcw/embed/entrypoint is only built on x86_64, but
internal/mkcw/embed/entrypoint.gz is run on all arches. This causes build
failures on anything non x86_64 as internal/mkcw/embed/entrypoint is not build.

Signed-off-by: Dan Čermák <dcermak@suse.com>
---
 Makefile | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 85b43c7b9..cce615d5f 100644
--- a/Makefile
+++ b/Makefile
@@ -73,17 +73,19 @@ bin/buildah: $(SOURCES) cmd/buildah/*.go internal/mkcw/embed/entrypoint.gz
 	$(GO_BUILD) $(BUILDAH_LDFLAGS) $(GO_GCFLAGS) "$(GOGCFLAGS)" -o $@ $(BUILDFLAGS) ./cmd/buildah
 
 ifneq ($(shell as --version | grep x86_64),)
+internal/mkcw/embed/entrypoint.gz: internal/mkcw/embed/entrypoint
+	$(RM) $@
+	gzip -k $^
+
 internal/mkcw/embed/entrypoint: internal/mkcw/embed/entrypoint.s
 	$(AS) -o $(patsubst %.s,%.o,$^) $^
 	$(LD) -o $@ $(patsubst %.s,%.o,$^)
 	strip $@
 else
 .PHONY: internal/mkcw/embed/entrypoint
+.PHONY: internal/mkcw/embed/entrypoint.gz
 endif
 
-internal/mkcw/embed/entrypoint.gz: internal/mkcw/embed/entrypoint
-	$(RM) $@
-	gzip -k $^
 
 .PHONY: buildah
 buildah: bin/buildah
-- 
2.42.0

